flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -332,11 +359,24 @@ public void renameTable(ObjectPath tablePath, String newTableName, boolean ignor
 		checkNotNull(tablePath, "tablePath cannot be null");
 		checkArgument(!StringUtils.isNullOrWhitespaceOnly(newTableName), "newTableName cannot be null or empty");
 
+		ObjectPath newPath = new ObjectPath(tablePath.getDatabaseName(), newTableName);
+
+		ConnectorCatalogTable connectorTable = connectorTables.remove(tablePath);
+
+		if (connectorTable != null) {
+			if (connectorTables.containsKey(newPath)) {
+				throw new TableAlreadyExistException(getName(), newPath);
```

Comment:
If we get here because of this exception, the table is already dropped, which doesn't seem right.